data_structures/linked_list/singly_link_list_deletion.c

```diff
@@ -73,11 +88,11 @@ void viewlist()  // function to display values
 
 int main()
 {
-    int n;
+    int n,pos;
```

Comment:
Let us initialize the variables.
```suggestion
    int n = 0, pos = 0;
```